app/controllers/api/v1/bookmarks_controller.rb

```diff
@@ -11,13 +11,14 @@ def show; end
 
   # POST /api/v1/bookmarks
   def create
-    @bookmark = Bookmark.find_or_create_for_user user: current_user, uri_string: params.dig(:data, :attributes, :url)
+    @bookmark = Bookmark.for(current_user, params.dig(:data, :attributes, :url))
 
     authorize @bookmark
 
-    @bookmark.update bookmark_params
+    @bookmark.assign_attributes bookmark_params.to_h.compact
+    @bookmark.tags = @bookmark.tags.to_a.concat(tags_models)
```

Comment:
@robacarp how about this pattern? and below in `#update` it just overwrites the tags rather than merging them.

Reply:
I like it but `tags_models` is bad english